### PR TITLE
Using Coq 8.10 ssreflect new features

### PR DESCRIPTION
**Pull request by CohenCyril**
##### Motivation for this change

Since we dropped Coq 8.9, we can now use `+`, `/[apply]`, `/[dup]`, and `/[swap]` in intro patterns and the tactic `under`.
This PR could be merged as is or expanded incrementally.

##### Things done/to do

- ~added corresponding entries in `CHANGELOG_UNRELEASED.md` (do not edit former entries)~
- ~added corresponding documentation in the headers~

##### Automatic note to reviewers

Read [this Checklist](https://github.com/math-comp/math-comp/wiki/Checklist-for-following,-reviewing-and-playing-with-a-PR#checklist-for-reviewing-a-pr) and make sure there is a milestone.

### CODE REVIEW

File: mathcomp/algebra/matrix.v
Comment (gares) [inline mathcomp/algebra/matrix.v:2467]: ?

**Comment by gares** [comment]
@erikmd I think you may want to review this PR ;-)

File: mathcomp/algebra/matrix.v
Comment (CohenCyril) [reply mathcomp/algebra/matrix.v:2467]: I tried to replace, it was not a good instance, and I thought the script still made more sense like this.

**Comment by erikmd** [comment]
Hi @gares and @CohenCyril ! yes, I'd be glad to have a look :)

ideally, when would you like to merge the PR? (I guess I'd need to review it before your next mathcomp-dev meeting on 2 Dec. ?)

**Comment by CohenCyril** [comment]
Hi @erikmd I don't think there is a strong need to review this PR, I guess it can be merged as such (or wait until after the release), and further script improvements can always come in later on. Moreover the release is likely to occur next week (so that we can attend the CUDW with no pressure).

**Comment by erikmd** [comment]
Hi @CohenCyril, sure! it can be merged as is indeed (and if I happen to have a look after the merge, I could just as well open a small additional PR!) anyway, that's great to be able to rely on `under` now :)

File: mathcomp/ssreflect/seq.v
Comment (gares) [inline mathcomp/ssreflect/seq.v:3632]: This is my favorite!

**Review by gares — APPROVED** [review]

File: mathcomp/algebra/matrix.v
Comment (CohenCyril) [inline mathcomp/algebra/matrix.v:2001]: @gares This one is my favorite

File: mathcomp/algebra/matrix.v
Comment (CohenCyril) [inline mathcomp/algebra/matrix.v:1999]: @gares This one is my favorite

Comment (gares) [reply mathcomp/algebra/matrix.v:1999]: I see the gain in terms of script, but it still applies under in the goal's head, not under some extra context.
Well I guess anyone has the right to have a favorite ;-)

**Pull request merged**
